trace-sdk/src/androidTest/java/io/bitrise/trace/data/management/DataManagerInstrumentedTest.java

```diff
@@ -204,16 +209,25 @@ public void reset_shouldNotBeInitialised() {
 
     /**
      * When the {@link DataManager#handleReceivedData(Data)} is called with a valid {@link Data} that will be
-     * converted to a {@link io.opencensus.proto.metrics.v1.Metric} it should not throw an exception.
+     * converted to a {@link io.opencensus.proto.metrics.v1.Metric} it should not throw an exception and be saved
+     * to the database correctly.
      */
     @Test
-    @UiThreadTest
```

Comment:
Why did you remove this annotation?

Reply:
Because I've updated the assertion to check the item now exists in the database, room won't work on the main thread. It gives this exception:
<img width="1424" alt="Screenshot 2021-05-10 at 19 11 47" src="https://user-images.githubusercontent.com/71286749/117705138-96b33c80-b1c3-11eb-9f88-56627d905071.png">

Is there a reason you think it should be there?

Reply:
Ah interesting problem. That explains why the test occasionally fails on CI. I've added the annotation back so that the test runs on it's own - but made the actual work on an `AsyncTask`. I'll add a comment to about why this works this way for future. Have added this in a0e7e82 .